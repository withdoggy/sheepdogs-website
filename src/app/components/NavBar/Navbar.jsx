import React, { Component } from 'react';
import { Menu, Dropdown } from 'semantic-ui-react';
import { Link } from 'react-router-dom';

export default class NavBar extends Component {
  state = {
    handleItemClick: this.handleItemClick,
    activeItem: '',
  };

  handleItemClick = (e, { name }) => this.setState({ activeItem: name });
  accountMenu = () => {
    if (this.props.currentUser === null) {
      return (
        <Dropdown.Item name="login" active={this.state.activeItem === 'login'} onClick={this.handleItemClick}>
          Zaloguj
        </Dropdown.Item>
      );
    } else {
      return (
        <>
          <Dropdown.Item name="login" active={this.state.activeItem === 'login'} onClick={this.handleItemClick}>
            Profil
          </Dropdown.Item>
          <Dropdown.Item name="logout" active={this.state.activeItem === 'logout'} onClick={this.handleItemClick}>
            Wyloguj
          </Dropdown.Item>
        </>
      );
    }
  };
  render() {
    return (
      <Menu fixed="top" inverted>
        <Menu.Item name="home" active={this.state.activeItem === 'home'} onClick={this.handleItemClick}>
          <Link to="/">Home</Link>
        </Menu.Item>
        <Menu.Item name="events" active={this.state.activeItem === 'events'} onClick={this.handleItemClick}>
          <Link to="/events">Events</Link>
        </Menu.Item>
        <Menu.Menu position="right">
          <Dropdown item text="Konto">
            <Dropdown.Menu>{this.accountMenu()}</Dropdown.Menu>
          </Dropdown>
        </Menu.Menu>
      </Menu>
    );
  }
}
