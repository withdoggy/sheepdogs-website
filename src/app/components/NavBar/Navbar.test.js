import React from 'react';
import renderer from 'react-test-renderer';

import NavBar from './Navbar';

describe('Navbar', () => {
  let props;

  beforeEach(() => {
    props = {
      currentUser: undefined,
    };
  });

  it('render navbar for anonymous user', () => {
    const NavBarComponent = renderer.create(<NavBar {...props} />).toJSON();
    expect(NavBarComponent).toMatchSnapshot();
  });
  it('render navbar for authenticated user', () => {
    let props = {
      currentUser: { username: 'Tolo' },
    };

    const NavBarComponent = renderer.create(<NavBar {...props} />).toJSON();
    expect(NavBarComponent).toMatchSnapshot();
  });
});
