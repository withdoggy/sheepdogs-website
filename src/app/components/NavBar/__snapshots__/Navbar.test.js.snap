// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar render navbar for anonymous user 1`] = `
<div
  className="ui inverted top fixed menu"
>
  <a
    className="item"
    onClick={[Function]}
  >
    Home
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    Events
  </a>
  <div
    className="right menu"
  >
    <div
      aria-expanded={false}
      className="ui item dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        Konto
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={false}
          className="item"
          name="login"
          onClick={[Function]}
          role="option"
        >
          Profil
        </div>
        <div
          aria-checked={false}
          className="item"
          name="logout"
          onClick={[Function]}
          role="option"
        >
          Wyloguj
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Navbar render navbar for authenticated user 1`] = `
<div
  className="ui inverted top fixed menu"
>
  <a
    className="item"
    onClick={[Function]}
  >
    Home
  </a>
  <a
    className="item"
    onClick={[Function]}
  >
    Events
  </a>
  <div
    className="right menu"
  >
    <div
      aria-expanded={false}
      className="ui item dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        Konto
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={false}
          className="item"
          name="login"
          onClick={[Function]}
          role="option"
        >
          Profil
        </div>
        <div
          aria-checked={false}
          className="item"
          name="logout"
          onClick={[Function]}
          role="option"
        >
          Wyloguj
        </div>
      </div>
    </div>
  </div>
</div>
`;
